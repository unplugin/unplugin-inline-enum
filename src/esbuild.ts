/**
 * This entry file is for esbuild plugin.
 *
 * @module
 */

import { InlineEnum } from './index'

/**
 * Esbuild plugin
 *
 * @example
 * ```ts
 * // esbuild.config.js
 * import { build } from 'esbuild'
 *
 * build({
 *   plugins: [require('unplugin-inline-enum/esbuild')()],
 * })
 * ```
 */
const esbuild = InlineEnum.esbuild as typeof InlineEnum.esbuild
export default esbuild
export { esbuild as 'module.exports' }
