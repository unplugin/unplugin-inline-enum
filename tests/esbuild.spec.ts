import path from 'node:path'
import { build } from 'esbuild'
import { expect, test } from 'vitest'
import UnpluginInlineEnum from '../src/esbuild'

test('esbuild', async () => {
  const result = await build({
    entryPoints: [path.resolve(__dirname, 'fixtures/main.ts')],
    bundle: true,
    write: false,
    format: 'esm',
    plugins: [
      UnpluginInlineEnum({
        scanDir: path.resolve(__dirname, 'fixtures'),
        scanMode: 'fs',
      }),
    ],
  })
  expect(result.outputFiles[0].text).toMatchSnapshot()
})
