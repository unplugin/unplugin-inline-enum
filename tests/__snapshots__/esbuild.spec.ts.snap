// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`esbuild 1`] = `
"(() => {
  // tests/fixtures/mod.ts
  var values = ["foo", 100, 4];

  // tests/fixtures/main.ts
  console.log("foo", 100, 4);
  console.log(values);
})();
"
`;
